qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -863,18 +862,23 @@ def analysis_results(
         self._retrieve_analysis_results(refresh=refresh)
         if index is None:
             return self._analysis_results.values()
+
+        def make_not_found_message(index):
```

Comment:
Should this be private?
```suggestion
        def _make_not_found_message(index: int) -> str:
```

Reply:
It doesn't hurt anything to do that. The function is only defined within the method though so it is already private. I will make the change separately from your suggestion because the type hint for `index` needs to be `Union[int, slice, str]`.

Reply:
```suggestion
        def _make_not_found_message(index: Union[int, slice, str]) -> str:
```